src/dmd/dmangle.d

```diff
@@ -170,9 +169,8 @@ private extern (C++) final class Mangler : Visitor
 {
     alias visit = Visitor.visit;
 public:
-    static assert(Key.sizeof == size_t.sizeof);
-    AA* types;
-    AA* idents;
+    size_t[void*] types;
+    size_t[void*] idents;
```

Comment:
Why is `void*` used instead of the actual types inserted in the AA?

Reply:
Found the answer in another PR.

Reply:
Also Walter explained in the linked PR.

Reply:
That it is an optimization is not too obvious unless you know. ☺️

Reply:
> That it is an optimization is not too obvious unless you know.

That's not just an optimization, but a necessity as using C++ classes as key just crashes when trying to access the type info.

Reply:
Perhaps that should be an error instead?

Reply:
>Perhaps that should be an error instead?

Or just fall back to void* typeinfo. See also https://issues.dlang.org/show_bug.cgi?id=13875 and https://issues.dlang.org/show_bug.cgi?id=15813.